File: backend/functions/src/features/authentication/domain/entities/onboarding_step.ts
import {z} from "zod";

/**
 * Etapa atual do onboarding (fonte da verdade no servidor após autenticação).
 */
export const onboardingStepSchema = z.enum([
  "none",
  "emailOtpPending",
  "profileComplete",
  "selfie",
  "liveness",
  "verification",
  "permissions",
  "done",
]);

export type OnboardingStep = z.infer<typeof onboardingStepSchema>;

File: app/lib/core/enums/authentication/onboarding_step.dart
import 'package:dart_mappable/dart_mappable.dart';

part 'onboarding_step.mapper.dart';

/// Etapa atual do onboarding (fonte da verdade no servidor após autenticação).
@MappableEnum()
enum OnboardingStep {
  /// Ainda sem fluxo definido (ex.: recém-criado no cliente antes do primeiro sync).
  none,

  /// Dados básicos de cadastro concluídos (nome, e-mail, etc.).
  profileComplete,

  /// Prova de vida (SDK / sessão).
  liveness,

  /// Fluxo de permissões (notificação, local, biometria).
  permissions,

  /// Onboarding finalizado; app pode ir para home.
  done,
}
